(FM-7550) changing test_harness_run logic

Thomas-Franklin opened the pull request:
Previously `test_harness_dependencies` would get called from an existing class without having relevancy to the current test, which causes unexpected behaviour.

By passing it where required, it should also increase the test performance as it is no longer performing the `test_harness_dependencies` from interface lib on every test case.

Commit:
(FM-7550) changing test_harness_run logic

Previously `test_harness_dependencies` would get called from an existing class without having relevancy to the current test, which causes unexpected behaviour.

By passing it where required, it should also increase the test performance as it is no longer performing the `test_harness_dependencies` from interface lib on every test case.

DavidS commented on tests/beaker_tests/lib/utilitylib.rb:
```diff
@@ -884,7 +875,7 @@ def supported_property_hash(tests, id, property_hash)
 # - Creates puppet resource title strings
 # - Cleans resource
-def test_harness_run(tests, id, skip_idempotence_check=false)
+def test_harness_run(tests, id, harness_dependencies=nil, skip_idempotence_check=false)
```
to avoid the issues around renumbering arguments, you should use named params:

```ruby
def test_harness_run(tests, id, harness_dependencies: nil, skip_idempotence_check: false)
```

DavidS commented on tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb:
```diff
@@ -337,10 +340,10 @@ def version_unsupported_properties(_tests, _id)
+  test_harness_run(tests, :suppress_uuc_true, TestVxLanVtepVni.method(:test_harness_dependencies))
+  test_harness_run(tests, :suppress_uuc_false, TestVxLanVtepVni.method(:test_harness_dependencies))
+  test_harness_run(tests, :multisite_ingress_replication_true, TestVxLanVtepVni.method(:test_harness_dependencies))
+  test_harness_run(tests, :multisite_ingress_replication_false, TestVxLanVtepVni.method(:test_harness_dependencies))
```
Looking at this, I more and more don't like it. There is too much duplication with the repeated `method()` calls, and it doesn't look like good ruby.

Please rework this instead to pass in the class itself, and have the `test_harness_run` method call the right thing:

```ruby
test_harness_run(tests, :multisite_ingress_replication_false, TestVxLanVtepVni)
```

and

```ruby
def test_harness_run(tests, id, harness_class=nil, skip_idempotence_check=false)
  # ...
  harness_class.test_dependencies(tests, id) if harness_class
```

Commit:
(FM-7550) extraction of unsupported_properties to a class

Similar to the `test_harness_dependencies` both `unsupported_properties` and `version_unsupported_properties` had the same behaviour, these changes should mean that these `magical` methods are contained in the test that requires them.

Thomas-Franklin commented:
Using this branch on the adhoc pipeline to fix up failures as they come.

DavidS reviewed (approved):
The new approach looks much more readable to me
